src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.Fixer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+
+namespace Microsoft.CodeQuality.Analyzers.Maintainability
+{
+    // TODO summary
+    /// <summary>
+    /// 
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.VisualBasic, LanguageNames.CSharp), Shared]
+    public class UseNameOfInPlaceOfStringFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(UseNameofInPlaceOfStringAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers'
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(CancellationToken.None);
+            var diagnostics = context.Diagnostics;
+            var diagnosticSpan = diagnostics.First().Location.SourceSpan;
+            // getInnerModeNodeForTie = true so we are replacing the string literal node and not the whole argument node
+            var nodeToReplace = root.FindNode(diagnosticSpan, getInnermostNodeForTie: true);
+
+            if (nodeToReplace != null)
+            {   
+                var stringText = diagnostics.First().Properties[UseNameofInPlaceOfStringAnalyzer.StringText];
+                context.RegisterCodeFix(
+                    CodeAction.Create(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringTitle, c => ReplaceWithNameOf(context.Document, nodeToReplace, stringText, c), equivalenceKey: nameof(UseNameOfInPlaceOfStringFixer)),
+                    context.Diagnostics);
+            }
+        }
+
+        private async Task<Document> ReplaceWithNameOf(Document document, SyntaxNode nodeToReplace, string stringText, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+
+            var nameOfExpression = generator.NameOfExpression(generator.IdentifierName(stringText));
```

Comment:
I assume that the generator handles escaping things properly, but can you make sure you have a test with a reserved keyword as the stringText?

Reply:
Yes, test has been added

Reply:
I don't see a test for this comment?

---
In reply to: [153680413](https://github.com/dotnet/roslyn-analyzers/pull/1367#discussion_r153680413) [](ancestors = 153680413)

Reply:
This test was added to UseNameofInPlaceOfStringTests.cs:

```csharp
class C
{
    void M(int x)
    {
        throw new ArgumentNullException("static");
    }
}`

Is that what you meant?